Week3/homework/3-step3.js

```diff
@@ -2,24 +2,37 @@
 
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
-  // add your code here
-  return str;
+  
+  let answerStr = "";
+  for(let i = 0; i < num; i++) {
+    answerStr += str;
+  }
+  return answerStr;
 }
-
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
+
+
 // use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // add your code here
-  return str;
+  let ansStr = "";
+  while(num > 0){
+    ansStr += str;
+    num--;
+  }
+  return ansStr;
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
 // use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // add your code here
-  return str;
-}
-
-console.log('while', repeatStringNumTimesWithDoWhile('abc', 3));
+  let repeat = "";
+  if(num < 0) return ""; //I used this to pass the challenge on Freecodecamp
```

Comment:
Aahaan. I wonder why you didn't use this logic in for and while loop. Also, you can simply return `repeat` variable from the if block as it itself is an empty string.

Reply:
@yash-kapila  Yeah like I commented in the other loops.. I forgot to put them, this was a challenge on freecodecamp, and with the others it passed me the challenge, but with this last when I builded my do while loop it throw me the error of if num was < 0 so I came up with the empty string return and forgot to add it to the others